src/index.d.ts

```diff
@@ -3843,6 +3843,63 @@ declare namespace esb {
         field?: string
     ): AvgAggregation;
 
+    /**
+     * A single-value metrics aggregation that computes the weighted average of numeric values that are extracted from the aggregated documents.
+     * These values can be extracted either from specific numeric fields in the documents.
+     * 
+     * [Elasticsearch reference](https://www.elastic.co/guide/en/elasticsearch/reference/current/search-aggregations-metrics-weight-avg-aggregation.html)
+     * 
+     * Added in Elasticsearch v6.4.0
+     * [Release notes](https://www.elastic.co/guide/en/elasticsearch/reference/6.4/release-notes-6.4.0.html)
+     *
+     * @param {string} name The name which will be used to refer to this aggregation.
+     * @param {string=} value The field or script to be used as the value.
+     * @param {string | Script =} weight The field or script to be used as the weighting.
+     * @extends MetricsAggregationBase
+     */
+    export class WeightedAverageAggregation extends MetricsAggregationBase {
```

Comment:
Could you add overrides similar to https://github.com/sudo-suhas/elastic-builder/blob/v2.26.0/src/index.d.ts#L627-L631 for the methods that are inherited from `MetricsAggregationBase` but cannot be called on `WeightedAverageAggregation`?

Reply:
Added